app/controllers/workflow_service_controller.rb

```diff
@@ -0,0 +1,80 @@
+class WorkflowServiceController < ApplicationController
+  ##
+  # Is a document closeable?
+  def closeable
+    @status = check_if_can_close_version
+    render json: @status
+  end
+
+  ##
+  # Is a document openable?
+  def openable
+    @status = check_if_can_open_verison
+    render json: @status
+  end
+
+  ##
+  # Has an object been published?
+  def published
+    @status = check_if_published
+    render json: @status
+  end
+
+  ##
+  # Has an object been submitted?
+  def submitted
+    @status = check_if_submitted
+    render json: @status
+  end
+
+  ##
+  # Has an object been accessioned?
+  def accessioned
+    @status = check_if_accessioned
+    render json: @status
+  end
+
+  private
+
+  ##
+  # Ported over logic from app/helpers/dor_object_helper.rb#LN119
+  # @return [Boolean]
+  def check_if_published
+    return true if Dor::WorkflowService.get_lifecycle('dor', params[:pid], 'published')
```

Comment:
Although I like the reuse idea, I disagree with the approach of putting it into a helper. I think helpers should be used for sharing code between controllers and views, and by adding these methods to helpers, it makes it easier for dev's to put these blocking requests into a view, causing performance issues.

Is a class under `lib` an approach that would be useful?